src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    private String color;
+    private double area;
+
+    public Figure() {
+        color = new ColorSupplier().getRandomColor();
+    }
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public double getArea() {
+        return area;
+    }
+
+    public void setArea(int area) {
```

Comment:
The `setArea` method should accept a `double` parameter instead of `int` to accommodate more precise area values.